integrations/height/save.go

```diff
@@ -0,0 +1,116 @@
+package height
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"go.uber.org/zap"
+
+	"go.autokitteh.dev/autokitteh/integrations"
+	"go.autokitteh.dev/autokitteh/sdk/sdkintegrations"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+// handleSave saves connection variables for an AutoKitteh connection.
+// This may result in a fully-initialized and usable connection, or it
+// may be an intermediate step before starting a 3-legged OAuth 2.0 flow.
+// This handler accepts both GET and POST requests alike. Why GET? This
+// is the only option when the web UI opens a pop-up window for OAuth.
+func (h handler) handleSave(w http.ResponseWriter, r *http.Request) {
+	c, l := sdkintegrations.NewConnectionInit(h.logger, w, r, desc)
+
+	// Check the "Content-Type" header in POST requests.
+	contentType := r.Header.Get("Content-Type")
+	expected := "application/x-www-form-urlencoded"
+	if r.Method == http.MethodPost && !strings.HasPrefix(contentType, expected) {
+		l.Warn("save connection: unexpected POST content type", zap.String("content_type", contentType))
+		c.AbortBadRequest("unexpected request content type")
+		return
+	}
+
+	// Parse GET request's query params / POST request's body.
+	if err := r.ParseForm(); err != nil {
+		l.Warn("save connection: failed to parse HTTP request", zap.Error(err))
+		c.AbortBadRequest("request parsing error")
+		return
+	}
+
+	// Sanity check: the connection ID is valid.
+	cid, err := sdktypes.StrictParseConnectionID(c.ConnectionID)
+	if err != nil {
+		l.Warn("save connection: invalid connection ID", zap.Error(err))
+		c.AbortBadRequest("invalid connection ID")
+		return
+	}
+
+	// Determine what to save and how to proceed.
+	vsid := sdktypes.NewVarScopeID(cid)
+	authType := h.saveAuthType(r.Context(), vsid, r.FormValue("auth_type"))
+
+	switch authType {
+	// Use the AutoKitteh's server's default Linear OAuth 2.0 app, i.e.
```

Comment:
Typo my Linear code that I didn't notice until now
```suggestion
	// Use the AutoKitteh server's default Linear OAuth 2.0 app, i.e.
```